### PR TITLE
Implement async

### LINKED ISSUE #20: Uniformize options names
It would be nice at some point to move all config values to ``AWS_`` or ``S3_`` prefix to prevent collisions.

### PR DESCRIPTION
**Pull request by Bladrak**
Fix #20 #21 #23

- [x] Make region configurable
- [x] Wait on PR https://github.com/thumbor/thumbor/pull/575
- [x] Wait on new thumbor's release

### CODE REVIEW

**Comment by Bladrak** [comment]
@masom would you mind reviewing this? I'm not fully at ease with tornado's async. Maybe we should have another branch as well, given the BC Breaks this implies?

File: tc_aws/aws/bucket.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        self._bucket = bucket
+
+    @return_future
+    def get(self, path, callback=None):
+        """
+        Returns object at given path
+        :param path:
+        :param callback:
+        """
+        session = Botocore(service='s3', region_name='eu-west-1', operation='GetObject')
```
Comment (masom) [inline tc_aws/aws/bucket.py]: The `region_name` should probably be configurable?

File: tc_aws/aws/bucket.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    def get_url(self, key, method='GET', expiry=3600, callback=None):
+        """
+        Generates the presigned url for given key & methods
+        :param key:
+        :param method:
+        :param expiry:
+        :param callback:
+        :return:
+        """
+        session = botocore.session.get_session()
```
Comment (masom) [inline tc_aws/aws/bucket.py]: Is there a reason in some places `session` is used and in other `Botocore(...)` ?

**Comment by masom** [comment]
Regions should probably be configurable either using the config file or using a url parameter.

**Comment by masom** [comment]
Other than that, looks good.

File: tc_aws/aws/bucket.py (hunk 2 above)
Comment (Bladrak) [inline tc_aws/aws/bucket.py]: We can't call the ``get_presigned_url`` from ``Botocore`` as it's not an API call, but a method within the ``botocore`` library

**Comment by Bladrak** [comment]
@masom updated, tests will be failing while the PR https://github.com/thumbor/thumbor/pull/575 hasn't been merged and a new thumbor's version has been released.

**Comment by masom** [comment]
:+1:

**Pull request merged**